genet/validate/network.py

```diff
@@ -35,3 +36,33 @@ def describe_graph_connectivity(G):
     # find number of connected subgraphs
     dict_to_return['number_of_connected_subgraphs'] = len(find_connected_subgraphs(G))
     return dict_to_return
+
+
+def zero_value(value):
+    return value in {0, '0', '0.0'}
+
+
+def negative_value(value):
+    if isinstance(value, str):
+        return '-' in value
+    return value < 0
+
+
+def infinity_value(value):
+    if isinstance(value, str):
+        return value in ['inf', '-inf']
+    return math.isinf(value)
+
+
+def fractional_value(value):
+    if isinstance(value, str):
+        return ('0.' in value) and (value != '0.0')
+    return 1 > value > 0
+
+
+LINK_ATTRIBUTE_VALIDATION_TOOLBOX = {
```

Comment:
This feels a little bit like an impoverished class. What made you go with a dict here?

Reply:
lol, I started with a function, but it didnt have any arguments and just returned a dictionary so that's why a dict, might do a quick dataclass here instead

Reply:
I was thinking more that this dict holds key/value pairs where the value is a validation function, right? So could you have a validation class that has all of those functions as methods? If you need different implementations of those methods, you could extend the class.